### PR TITLE
For comments

### PR DESCRIPTION
**Pull request by ShaheerCH**
For this PR
**Home page**
When the page loads, the webapp retrieves data from:

- MealsDB API
- The Involvement API to show the item likes.
- When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
- When the user clicks on the "Comments" button, the Comments popup appears.

- When the popup loads, the webapp retrieves data from:
- The selected API and shows details about the selected item.

- The Involvement API to show the item comments.
- When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,275 @@
...
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-use-before-define */
+
+import './index.css';
+
+// Define the constants
+const mealList = document.getElementById('meal');
+const mealDetailsContent = document.querySelector('.meal-details-content');
+const recipeCloseBtn = document.getElementById('.recipe-close-btn');
+const url = 'https://www.themealdb.com/api/json/v1/1/filter.php?i=egg';
```
Comment (mateo951) [inline src/index.js:10]: - The web app uses an API that bases its development around it ✔
- The API should allow for getting a list of items with a unique item id and detailed information about the items. ✔
 
Well done 🙌🔥

File: src/index.js [hunk 2]
```diff
@@ -1 +1,275 @@
...
+  }
+});
+
+const postLike = async (mealId) => {
+  const like = {
+    item_id: mealId,
+  };
+
+  const response = await fetch(
+    'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/bbDC3TOidzHVfwfLZkFs/likes',
```
Comment (mateo951) [inline src/index.js:183]: The project uses the Involvement API to record the different user interactions (likes and comments) 📝

File: dist/index.html [hunk 3]
```diff
@@ -1,12 +1,64 @@
...
+          <h2>Try Some Recipes</h2>
+
+          <!-- Meal-cards container  -->
+          <div id="meal"></div>
+        </div>
+
+        <!-- Popup detail- -->
+        <div class="meal-details hide"></div>
+
+        <!-- End of details popup -->
```
Comment (mateo951) [inline dist/index.html:52]: Popup is working beautifully ✔ Congrats 👌

File: src/index.js [hunk 4]
```diff
@@ -1 +1,275 @@
...
+const url = 'https://www.themealdb.com/api/json/v1/1/filter.php?i=egg';
+const modalDetail = document.querySelector('.meal-details');
+
+// Add event listeners
+let foodData = [];
+let likesData = [];
+const data2 = [];
+const orderedLikesData = [];
+
+const getFood = async () => {
```
Comment (mateo951) [inline src/index.js:19]: Nice use of arrow functions 🔥

**Review by mateo951 — CHANGES_REQUESTED** [review]
### Hi @ShaheerCH and @AbbyNyakara, 

📝👌 **Your project highlights** 😎 
- General Requirements 💪🤓 (Gitflow 🔥)
- UI Design 👌
- Implementation of API's 🤯

You have done a great job with your app! There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by mateo951** [comment]
❗ [REQUIRED] ❗
Please be sure to add a link to the video presentation in the code review summary 🤓

image

The video should address the following points:
1. The student recorded a video presenting the features of the project.
2. The maximum length of the video is 5 minutes.
3. The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
4. All the members in the video should appear in the video and talk about the project.
5. The video contains a highlight about a piece of the code the student finds interesting or is proud of making

[OPTIONAL] 
On another note, kindly be sure to add a descriptive PR with a nice format cause it makes your project more professional 👀

**Comment by AbbyNyakara** [comment]
@mateo951 , We have updated the project presentation in the README file, (under project demo link). Please review.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,95 @@
...
+
+![](https://img.shields.io/badge/Microverse-blueviolet)
+
+# RECIPE HUB PROJECT
+
+This is the capstone project for module two in Microverse (Kanban board).
+In this project we created a fully functional web app that fetches tv shows from an api [Meal App](https://www.themealdb.com)
+Users can also comment and like movies and shows.
+
+### [Project Demo Link](https://www.loom.com/share/e59aee87611948e5961a0c344acc8d61)
```
Comment (MahmoudBakr23) [inline README.md:13]: - Please add another link to the live demo link of the deployed version of the app besides the project presentation. Both links should be listed in the README file 🙏

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Well done making the required changes
- Nice presentation
- Good communication

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**